tony-core/src/main/java/com/linkedin/tony/events/EventHandler.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright 2018 LinkedIn Corporation. All rights reserved. Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.tony.events;
+
+import com.linkedin.tony.TonyJobMetadata;
+import com.linkedin.tony.util.HistoryFileUtils;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.concurrent.BlockingQueue;
+import org.apache.avro.file.DataFileWriter;
+import org.apache.avro.io.DatumWriter;
+import org.apache.avro.specific.SpecificDatumWriter;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.FileUtil;
+import org.apache.hadoop.fs.Path;
+
+
+public class EventHandler implements Runnable {
```

Comment:
Please add some tests for this EventHandler

Reply:
Added.